src/context/providers/DataProvider.js

```diff
@@ -0,0 +1,54 @@
+import { useReducer, useContext, createContext } from "react";
+
+import dataReducer from "../reducers/dataReducer";
+
+const dataContext = createContext();
+
+const initialState = {
+  vid: {
+    loading: false,
+    items: [],
+    error: null,
+    single_video: {},
```

Comment:
Can you explain why you keep states for loading and error? as a initial state

Reply:
If I keep a single loading, all pages will show loading - history, liked, playlist, that's why I separated the loading and error states.